tests/core/vm/test_vm.py

```diff
@@ -191,3 +194,55 @@ def test_validate_gas_limit_too_high(noproof_consensus_chain):
 
     with pytest.raises(ValidationError, match="[Gg]as limit"):
         vm.validate_header(invalid_header, block1.header)
+
+
+@pytest.mark.parametrize(
+    "custom_header_params,null_parent",
+    (
+        ({
+            'gas_limit': 1,
+            'difficulty': 10,
+            'timestamp': 100,
+            'block_number': 1000
+        }, True),
+        ({
+            'gas_limit': 1,
+            'difficulty': 10,
+            'timestamp': 100,
+        }, False),
+        ({
+            'gas_limit': 1,
+            'difficulty': 10,
+            'timestamp': 100,
+            'block_number': 1000,
+        }, False)
+    )
+)
+def test_fill_header_params_from_parent(custom_header_params, null_parent,
```

Comment:
The test cases look good here. We do try to avoid conditionals in our tests and instead split each test case out into their own test. With conditionals, it can be hard to read what test case ends where and to know which conditional is causing the test to pass over another without debugging the test, etc.

If we can split each test case out into its own test, with an explicit title of the thing it is testing, I think that would be ideal. We can probably factor out the parametrized test cases into a list and pass them into each test if they end up sharing similar info too... something like:

```py
CUSTOM_HEADER_TEST_DATA = (
    {
         'gas_limit': 1,
         'difficulty': 10,
         ...
    },
    { ... },
    { ... },
)

@pytest.mark.parametrize('custom_header_params', CUSTOM_HEADER_TEST_DATA)
def test_case_1(custom_header_params):
     ...
     
@pytest.mark.parametrize('custom_header_params', CUSTOM_HEADER_TEST_DATA)
def test_case_2(custom_header_params):
     ...
```

Thoughts?